sharding/handler/utils/log_parser.py

```diff
@@ -66,7 +66,7 @@ def _parse_value(self, *, val_type: str, val: bytes) -> Union[bool, str, bytes,
         if val_type == 'bool':
             return bool(big_endian_to_int(val))
         elif val_type == 'address':
-            return to_checksum_address(val[-20:])
+            return to_canonical_address(val[-20:])
```

Comment:
As this returns an `Address`, the type hint of the function should be updated (replace `str` with `Address`). Unfortunately, `eth_utils` does not use `eth_typing` yet, so they defined their own `Address` type. Therefore it might be necessary to `cast` their `Address into `eth_typing.Address`, I'm not sure though.